src/systems/rtti.rs

```diff
@@ -0,0 +1,312 @@
+use crate::raw::root::RED4ext as red;
+use crate::{Array, Bitfield, CName, Class, Enum, Function, Type};
+
+#[repr(transparent)]
+pub struct CRTTISystem(*mut red::CRTTISystem);
+
+impl CRTTISystem {
+    pub fn get() -> Self {
+        unsafe { Self(red::CRTTISystem_Get()) }
+    }
+
+    #[inline]
+    pub fn get_class(&self, name: CName) -> Option<&Class> {
+        let class = unsafe { (self.vft().base.get_class)(&(*self.0)._base, name.0) };
+        if class.is_null() {
+            return None;
+        }
+        Some(unsafe { &*class.cast::<Class>() })
+    }
+
+    #[inline]
+    pub fn get_type(&self, name: CName) -> Option<&Type> {
+        let ty = unsafe { (self.vft().base.get_type)(&(*self.0)._base, name.0) };
+        if ty.is_null() {
+            return None;
+        }
+        Some(unsafe { &*ty.cast::<Type>() })
+    }
+
+    #[inline]
+    pub fn get_enum(&self, name: CName) -> Option<&Enum> {
+        let ty = unsafe { (self.vft().base.get_enum)(&(*self.0)._base, name.0) };
+        if ty.is_null() {
+            return None;
+        }
+        Some(unsafe { &*ty.cast::<Enum>() })
+    }
+
+    #[inline]
+    pub fn get_bitfield(&self, name: CName) -> Option<&Bitfield> {
+        let ty = unsafe { (self.vft().base.get_bitfield)(&(*self.0)._base, name.0) };
+        if ty.is_null() {
+            return None;
+        }
+        Some(unsafe { &*ty.cast::<Bitfield>() })
+    }
+
+    #[inline]
+    pub fn get_function(&self, name: CName) -> Option<&Function> {
+        let ty = unsafe { (self.vft().base.get_function)(&(*self.0)._base, name.0) };
+        if ty.is_null() {
+            return None;
+        }
+        Some(unsafe { &*ty.cast::<Function>() })
+    }
+
+    #[inline]
+    pub fn get_native_types(&self) -> Vec<Type> {
```

Comment:
```suggestion
    pub fn get_native_types(&self) -> Array<Type> {
```
I think we can return the in-crate type here and not force the user to pay the cost of an allocation